Samples continuous tests failing due to fastavro upgrade

Continuous tests for samples have been failing consistently starting 10/03/2023. Error message:
```
capsys = <_pytest.capture.CaptureFixture object at 0x7fe400dcff70>
project_id = 'python-docs-samples-tests'

    def test_quickstart_wo_snapshot(capsys, project_id):
>       quickstart.main(project_id)

quickstart_test.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
quickstart.py:78: in main
    for row in rows:
[.nox/py-3-9/lib/python3.9/site-packages/google/cloud/bigquery_storage_v1/reader.py:348](https://cs.corp.google.com/piper///depot/google3/.nox/py-3-9/lib/python3.9/site-packages/google/cloud/bigquery_storage_v1/reader.py?l=348&ws&snapshot=0): in __iter__
    for row in page:
[.nox/py-3-9/lib/python3.9/site-packages/google/cloud/bigquery_storage_v1/reader.py:516](https://cs.corp.google.com/piper///depot/google3/.nox/py-3-9/lib/python3.9/site-packages/google/cloud/bigquery_storage_v1/reader.py?l=516&ws&snapshot=0): in next
    return next(self._iter_rows)
[.nox/py-3-9/lib/python3.9/site-packages/google/cloud/bigquery_storage_v1/reader.py:714](https://cs.corp.google.com/piper///depot/google3/.nox/py-3-9/lib/python3.9/site-packages/google/cloud/bigquery_storage_v1/reader.py?l=714&ws&snapshot=0): in to_rows
    yield fastavro.schemaless_reader(messageio, self._fastavro_schema)
[fastavro/_read.pyx:1138](https://cs.corp.google.com/piper///depot/google3/fastavro/_read.pyx?l=1138&ws&snapshot=0): in fastavro._read.schemaless_reader
    ???
[fastavro/_read.pyx:1165](https://cs.corp.google.com/piper///depot/google3/fastavro/_read.pyx?l=1165&ws&snapshot=0): in fastavro._read.schemaless_reader
    ???
[fastavro/_read.pyx:744](https://cs.corp.google.com/piper///depot/google3/fastavro/_read.pyx?l=744&ws&snapshot=0): in fastavro._read._read_data
    ???
[fastavro/_read.pyx:617](https://cs.corp.google.com/piper///depot/google3/fastavro/_read.pyx?l=617&ws&snapshot=0): in fastavro._read.read_record
    ???
[fastavro/_read.pyx:736](https://cs.corp.google.com/piper///depot/google3/fastavro/_read.pyx?l=736&ws&snapshot=0): in fastavro._read._read_data
    ???
[fastavro/_read.pyx:527](https://cs.corp.google.com/piper///depot/google3/fastavro/_read.pyx?l=527&ws&snapshot=0): in fastavro._read.read_union
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   EOFError

[fastavro/_read.pyx:177](https://cs.corp.google.com/piper///depot/google3/fastavro/_read.pyx?l=177&ws&snapshot=0): EOFError
```

It's likely caused by the same fastavro upgrade described in PR https://github.com/googleapis/python-bigquery-storage/pull/687.

--- a/samples/quickstart/quickstart.py
+++ b/samples/quickstart/quickstart.py
@@ -75,9 +75,14 @@ def main(project_id="your-project-id", snapshot_millis=0):
     names = set()
     states = set()
 
-    for row in rows:
-        names.add(row["name"])
-        states.add(row["state"])
+    # fastavro returns EOFError instead of StopIterationError starting v1.8.4.
+    # See https://github.com/googleapis/python-bigquery-storage/pull/687
+    try:
+        for row in rows:
+            names.add(row["name"])
+            states.add(row["state"])
+    except EOFError:
+        pass
 
     print("Got {} unique names in states: {}".format(len(names), ", ".join(states)))
     # [END bigquerystorage_quickstart]

